server.js

```diff
@@ -22,9 +17,80 @@ const app = express()
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+// --- ALL THE ROUTES ---
+
+// --- front page ---
+app.get('/', (request /* incoming */, response /* outgoing */) => {
+
+  response.send('Books API.')
+})
+
+
+// --- return ALL BOOKS & search by TITLE using QUERY SELECTORS ---
+//localhost:8080/books?title=freakonomics
+
+app.get("/books", (request, response) => {
+
+  const titleSearch = request.query.title
+
+  if (titleSearch === undefined) {
+    response.json(booksData)
+  }
+  else if (titleSearch !== undefined) {
+    const titleLowerCase = titleSearch.toLowerCase()
+
+    //this could be written more clear ...? 
+    const titleArray = booksData.filter(book => book.title.toString().toLowerCase().includes(titleLowerCase))
+
+    if (titleArray.length > 0) {
+      response.json(titleArray)
+    }
+    else if (titleArray.length === 0) {
+      response.status(404).json(`Ups, no keyword called '${titleLowerCase}' in a title.`)
+    }
+  }
+})
+
+
+// --- getting SPECIFIC book based on ISBN13 ---
+//localhost:8080/id/:isbn13
+
+app.get("/id/:isbn13", (request, response) => {
```

Comment:
Endpoints should be named after what they return and not how they are filtered. This should therefore be named `/books/:isbn13`

Reply:
👍